robottelo/config/virtwho.py

```diff
@@ -0,0 +1,499 @@
+
+        The default value will return if the look up option is not available.
+        The value will be cast using a callable if specified otherwise a string
+        will be returned.
+
+        :param section: Section to look for.
+        :param option: Option to look for.
+        :param default: The value that should be used if the option is not
+            defined.
+        :param cast: If provided the value will be cast using the cast
+            provided.
+        """
+        # First try to read from environment variable.
+        # [bugzilla]
+        # api_key=123456
+        # can be expressed as:
+        # $ export ROBOTTELO_BUGZILLA_API_KEY=123456
+        value = os.environ.get(f'ROBOTTELO_{section.upper()}_{option.upper()}')
+
+        try:
+            # If envvar does not exist then try from .properties file.
+            value = value or self.config_parser.get(section, option)
+            if cast is not None:
+                if cast is bool:
+                    value = self.cast_boolean(value)
+                elif cast is dict:
+                    value = self.cast_dict(value)
+                elif cast is list:
+                    value = self.cast_list(value)
+                elif cast is tuple:
+                    value = self.cast_tuple(value)
+                else:
+                    value = cast(value)
+        except (NoSectionError, NoOptionError):
+            value = default
+        return value
+
+    def has_section(self, section):
+        """Check if section is available."""
+        return self.config_parser.has_section(section)
+
+
+class FeatureSettings(object):
+    """Settings related to a feature.
+
+    Create a instance of this class and assign attributes to map to the feature
+    options.
+    """
+
+    def read(self, reader):
+        """Subclasses must implement this method in order to populate itself
+        with expected settings values.
+
+        :param reader: An INIReader instance to read the settings.
+        """
+        raise NotImplementedError('Subclasses must implement read method.')
+
+    def validate(self):
+        """Subclasses must implement this method in order to validade the
+        settings and raise ``ImproperlyConfigured`` if any issue is found.
+        """
+        raise NotImplementedError('Subclasses must implement validate method.')
+
+
+class EsxSettings(FeatureSettings):
+    """VirtWho settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(EsxSettings, self).__init__(*args, **kwargs)
+        # Hypervisor Information
+        self.hypervisor_type = None
+        self.hypervisor_server = None
+        self.hypervisor_username = None
+        self.hypervisor_password = None
+        self.guest = None
+        self.guest_port = None
+        self.guest_username = None
+        self.guest_password = None
+        # SKU Information
+        self.sku_vdc_physical = None
+        self.sku_vdc_virtual = None
```

Comment:
we may create a new section in virthwo.properties for SKUs:
[sku]
vdc_physical=
vdc_virtual=

Reply:
Yes, that is a good idea.